packages/vitest-pool-workers/src/worker/fetch-mock.ts

```diff
@@ -170,7 +170,19 @@ globalThis.fetch = async (input, init) => {
 					statusText: responseStatusText,
 					headers: responseHeaders,
 				});
+				const throwImmutableHeaderError = () => {
+					throw new TypeError("Can't modify immutable headers");
+				};
 				Object.defineProperty(response, "url", { value: url.href });
+				Object.defineProperty(response.headers, "set", {
+					value: throwImmutableHeaderError,
+				});
+				Object.defineProperty(response.headers, "append", {
+					value: throwImmutableHeaderError,
+				});
+				Object.defineProperty(response.headers, "delete", {
+					value: throwImmutableHeaderError,
+				});
```

Comment:
I think if you could use `defineProperties` instead to do all of this in one single go, for example
```ts
const immutableValue = { value: throwImmutableHeaderError };
Object.defineProperties(response.headers, {
	set: immutableValue,
	append: immutableValue,
	delete: immutableValue,
});
```

Reply:
Good call! Updated in https://github.com/cloudflare/workers-sdk/pull/8254/commits/e54d234c1a71e5d23dd9b985e5a76e39f3d8763f